File: sampleapplication/src/main/java/com/byoutline/sampleapplication/MainActivity.java
package com.byoutline.sampleapplication;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.IdRes;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

import com.byoutline.sampleapplication.customfontviews.CustomFontViewsActivity;
import com.byoutline.sampleapplication.databinding.DataBindingActivity;
import com.byoutline.sampleapplication.draweractivity.ExampleActivity;
import com.byoutline.sampleapplication.networkchangereceiver.NetworkChangeActivity;
import com.byoutline.sampleapplication.roundedimageview.DrawableActivityExample;
import com.byoutline.sampleapplication.rx.RxLifecycleActivity;
import com.byoutline.sampleapplication.waitlayout.WaitActivity;
import com.byoutline.secretsauce.activities.WebViewActivityV7;

import java.util.HashMap;
import java.util.Map;

public class MainActivity extends AppCompatActivity {

    private final Map<Integer, Class<? extends AppCompatActivity>> btnClassMap = new HashMap<Integer, Class<? extends AppCompatActivity>>() {{
        put(R.id.baseAppCompatActivityBtn , ExampleActivity.class);
        put(R.id.networkChangeReceiverBtn , NetworkChangeActivity.class);
        put(R.id.roundedBtn , DrawableActivityExample.class);
        put(R.id.customViewsBtn , CustomFontViewsActivity.class);
        put(R.id.waitLayoutBtn , WaitActivity.class);
        put(R.id.rxBtn , RxLifecycleActivity.class);
        put(R.id.databindingBtn , DataBindingActivity.class);
    }};

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void onClick(View view) {
        Intent intent = getActivityIntentForView(view.getId());
        if (intent != null) {
            startActivity(intent);
        }
    }

    @Nullable
    private Intent getActivityIntentForView(@IdRes int viewId) {
        if(viewId == R.id.webVievActivityBtn) {
            // Remember about adding internet permission to manifest
            // and adding WebViewActivityV7 to manifest file.
            Intent intent = new Intent(this, WebViewActivityV7.class);
            intent.putExtra(WebViewActivityV7.BUNDLE_URL, "https://github.com/byoutline/SecretSauce");
            return intent;
        }
        if(btnClassMap.containsKey(viewId)) {
            return new Intent(this, btnClassMap.get(viewId));
        }
        return null;
    }
}

File: sampleapplication/src/main/java/com/byoutline/sampleapplication/MainActivity.kt
package com.byoutline.sampleapplication

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.byoutline.sampleapplication.databinding.ActivityMainBinding
import com.byoutline.secretsauce.activities.WebViewActivityV7
import com.byoutline.secretsauce.databinding.bindContentView
import com.byoutline.secretsauce.lifecycle.AttachableViewModel
import com.byoutline.secretsauce.lifecycle.getVMWithAutoLifecycle
import me.tatarka.bindingcollectionadapter2.ItemBinding
import javax.inject.Inject
import javax.inject.Qualifier

class MainActivity : ClassNameAsToolbarTitleActivity(), MainActivityItemClickListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        bindContentView<ActivityMainBinding>(R.layout.activity_main).viewModel =
            getVMWithAutoLifecycle(MainActivityVM::class)
    }

    override fun onItemClick(item: MainActivityItem) {
        val intent = Intent(this, item.clazz)
        if (item.clazz == WebViewActivityV7::class.java) {
            intent.putExtra(WebViewActivityV7.BUNDLE_URL, "https://github.com/byoutline/SecretSauce")
        }
        startActivity(intent)
    }
}

class MainActivityVM @Inject constructor(
    @MainActivityItems val items: List<MainActivityItem>
) : AttachableViewModel<MainActivityItemClickListener>() {
    private val listener: MainActivityItemClickListener = object : MainActivityItemClickListener {
        override fun onItemClick(item: MainActivityItem) {
            view?.onItemClick(item)
        }
    }
    val itemBinding: ItemBinding<MainActivityItem> =
        ItemBinding.of<MainActivityItem>(BR.item, R.layout.main_activity_item)
            .bindExtra(BR.listener, listener)
}

@Qualifier
@Retention(AnnotationRetention.RUNTIME)
internal annotation class MainActivityItems

interface MainActivityItemClickListener {
    fun onItemClick(item: MainActivityItem)
}

data class MainActivityItem(val clazz: Class<out AppCompatActivity>, private val name: String) {
    fun displayName() = "$name (${clazz.simpleName})"
}